### PR TITLE
Add ML workspace access functions

### PR DESCRIPTION
**Pull request by wmudge**
Signed-off-by: Webster Mudge <wmudge@cloudera.com>

### CODE REVIEW

File: src/cdpy/ml.py [hunk 1]
```diff
@@ -37,3 +37,52 @@ def describe_all_workspaces(self, env=None):
...
         return resp
+    
+    def list_workspace_access(self, name: str = None, crn: str = None, env: str = None):
+        resp = self.sdk.call(
+            svc='ml', func='list_workspace_access', ret_field='users', ret_error=True,
+            workspaceName=name,
+            environmentName=env,
+            workspaceCrn=crn
+        )
+        if isinstance(resp, CdpError):
```
Comment (Chaffelson) [inline src/cdpy/ml.py:48]: These should be handled with Squash I think, unless it cannot understand these particular errors?

File: src/cdpy/ml.py [hunk 2]
```diff
@@ -37,3 +37,52 @@ def describe_all_workspaces(self, env=None):
...
+
+    def grant_workspace_access(self, identifier: str, name: str = None, crn: str = None, env: str = None):
+        resp = self.sdk.call(
+            svc='ml', func='grant_workspace_access', ret_error=True,
+            workspaceName=name,
+            environmentName=env,
+            workspaceCrn=crn,
+            identifier=identifier
+        )
+        if isinstance(resp, CdpError):
```
Comment (Chaffelson) [inline src/cdpy/ml.py:65]: More squash

File: src/cdpy/ml.py [hunk 3]
```diff
@@ -37,3 +37,52 @@ def describe_all_workspaces(self, env=None):
...
+        
+    def revoke_workspace_access(self, identifier: str, name: str = None, crn: str = None, env: str = None):
+        resp = self.sdk.call(
+            svc='ml', func='revoke_workspace_access', ret_error=True,
+            workspaceName=name,
+            environmentName=env,
+            workspaceCrn=crn,
+            identifier=identifier
+        )
+        if isinstance(resp, CdpError):
```
Comment (Chaffelson) [inline src/cdpy/ml.py:82]: Fruit squash

File: src/cdpy/ml.py [hunk 4]
```diff
@@ -37,3 +37,52 @@ def describe_all_workspaces(self, env=None):
...
+            if resp.error_code == 'UNKNOWN':
+                self.sdk.throw_warning(CdpWarning(str(resp.violations)))
+            if resp.error_code == 'INVALID_ARGUMENT':
+                resp.update(message=resp.violations)
+                self.sdk.throw_error(resp)
+        return None
```
Comment (Chaffelson) [inline src/cdpy/ml.py:88]: newline at end of files please

**Review by Chaffelson — CHANGES_REQUESTED** [review]

File: src/cdpy/ml.py (hunk 1 above)
Comment (wmudge) [reply src/cdpy/ml.py:48]: I believe I can convert the first "catch" to use Squelch, i.e. `UNKNOWN`, but for the second error, i.e. `INVALID_ARGUMENTS`, I want the error to be thrown -- honor the `ret_error` -- but update the error `messages` with something more descriptive before returning the error itself.

**Pull request merged**